client/client.go

```diff
@@ -774,19 +775,26 @@ func (r *NotaryRepository) Update(forWrite bool) (*tufclient.Client, error) {
 // we should always attempt to contact the server to determine if the repository
 // is initialized or not. If set to true, we will always attempt to download
 // and return an error if the remote repository errors.
-func (r *NotaryRepository) bootstrapClient(checkInitialized bool) (*tufclient.Client, error) {
-	var (
-		rootJSON   []byte
-		err        error
-		signedRoot *data.SignedRoot
-	)
+func (r *NotaryRepository) bootstrapClient(checkInitialized bool) (tuf.RepoBuilder, error) {
+	var successfullyBootstrapped bool
+	version := 0
+	b := tuf.NewRepoBuilder(r.CertStore, r.gun, r.CryptoService, nil, nil)
+
 	// try to read root from cache first. We will trust this root
 	// until we detect a problem during update which will cause
 	// us to download a new root and perform a rotation.
 	rootJSON, cachedRootErr := r.fileStore.GetMeta("root", -1)
-
 	if cachedRootErr == nil {
-		signedRoot, cachedRootErr = r.validateRoot(rootJSON)
+		cachedRootErr = b.Load(data.CanonicalRootRole, rootJSON, version)
+		if cachedRootErr == nil {
+			successfullyBootstrapped = true
+		} else {
+			// try to get the version if we can
+			signedMeta := &data.SignedMeta{}
+			if err := json.Unmarshal(rootJSON, signedMeta); err == nil {
```

Comment:
Good point - it definitely failed validation at this point - I guess we really want to say "validate everything except expiry" and load the version otherwise?